### ISSUE TITLE
Arguments for pd.concat missing Series

### ISSUE THREAD

**Issue by Dr-Irv** [microsoft/pylance-release#390]
## Environment data

-   Language Server version: 2020.9.5
-   OS and version: Windows 10
-   Python version (& distribution if applicable, e.g. Anaconda): Anaconda Python 3.7.5

## Expected behaviour

No error

## Actual behaviour

```python
s1 = pd.Series([1, 2, 3])
s2 = pd.Series([4, 5, 6])
dc = pd.concat([s1, s2], axis=1)
print(dc)
```
reports
```
Argument of type "List[Series[Unknown]]" cannot be assigned to parameter "objs" of type "Iterable[DataFrame] | Mapping[Hashable | None, DataFrame]" in function "concat"
  Type "List[Series[Unknown]]" cannot be assigned to type "Iterable[DataFrame] | Mapping[Hashable | None, DataFrame]"
    "List[Series[Unknown]]" is incompatible with "Mapping[Hashable | None, DataFrame]"
      TypeVar "_T_co" is covariant
        "Series[Unknown]" is incompatible with "DataFrame"
```

Fix is:
```
diff concat.pyi.ORIG concat.pyi.IRV
1a2
> from pandas import Series as Series
4c5
< def concat(objs: Union[Iterable[DataFrame], Mapping[Optional[Hashable], DataFrame]], axis=..., join: str=..., ignore_index: bool=..., keys=..., levels=..., names=..., verify_integrity: bool=..., sort: bool=..., copy: bool=...) -> DataFrame: ...
---
> def concat(objs: Union[Iterable[DataFrame], Iterable[Series], Mapping[Optional[Hashable], DataFrame]], axis=..., join: str=..., ignore_index: bool=..., keys=..., levels=..., names=..., verify_integrity: bool=..., sort: bool=..., copy: bool=...) -> DataFrame: ...
```

**Comment by Dr-Irv** [comment]
It's actually missing combinations of `Series` and `DataFrame`, so this seems to be the right fix:
```
diff core/reshape/concat.pyi.ORIG core/reshape/concat.pyi
1a2
> from pandas import Series as Series
4c5
< def concat(objs: Union[Iterable[DataFrame], Mapping[Optional[Hashable], DataFrame]], axis=..., join: str=..., ignore_index: bool=..., keys=..., levels=..., names=..., verify_integrity: bool=..., sort: bool=..., copy: bool=...) -> DataFrame: ...
---
> def concat(objs: Union[Iterable[Union[DataFrame, Series]], Mapping[Optional[Hashable], DataFrame]], axis=..., join: str=..., ignore_index: bool=..., keys=..., levels=..., names=..., verify_integrity: bool=..., sort: bool=..., copy: bool=...) -> DataFrame: ...
```

**Comment by jakebailey** [comment]
This issue has been fixed in version 2020.9.6, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#202096-23-september-2020
